frontend/src/metabase/components/InputBlurChange/InputBlurChange.tsx

```diff
@@ -37,13 +46,25 @@ const InputBlurChange = ({
   const handleBlur = useCallback(
     (event: React.ChangeEvent<HTMLInputElement>) => {
       if (onBlurChange && (value || "") !== event.target.value) {
-        onBlurChange(event);
+        onBlurChange(event.target.value);
       }
     },
     [value, onBlurChange],
   );
 
-  const inputProps = _.omit(props, "onBlur", "onBlurChange", "onChange");
+  useEffect(() => {
```

Comment:
I think you could use `useLatest` from react-use https://github.com/streamich/react-use/blob/master/docs/useLatest.md and it would greatly simplify the logic here. Something like
```
const latestValueRef = useLatest(value);

useUnmount(() => {
  if (latestValueRef.current !== inputRef.value) {
    onBlurChange(inputRef.value)
  }
});
```